rust/operator-binary/src/crd/mod.rs

```diff
@@ -393,6 +402,34 @@ impl v1alpha1::AirflowCluster {
         fragment::validate(conf_rolegroup).context(FragmentValidationFailureSnafu)
     }
 
+    pub fn merged_listener_class(
+        &self,
+        role: &AirflowRole,
+        rolegroup_name: &String,
+    ) -> Option<String> {
+        if role == &AirflowRole::Webserver {
+            if let Some(webservers) = self.spec.webservers.as_ref() {
+                let conf_defaults = Some("cluster-internal".to_string());
+                let mut conf_role = webservers.config.config.listener_class.to_owned();
+                let mut conf_rolegroup = webservers
+                    .role_groups
+                    .get(rolegroup_name)
+                    .map(|rg| rg.config.config.listener_class.clone())
+                    .unwrap_or_default();
+
+                conf_role.merge(&conf_defaults);
+                conf_rolegroup.merge(&conf_role);
```

Comment:
Hm, don't love how this ends up bypassing "the main merge".. will have to look into how hdfs-op is doing it. Probably not worth blocking the PR over, more of a note for later.

Reply:
I'm fine with punting on this until Druid, but as long as we have a note to come back to Airflow eventually rather than letting it all diverge forever.